### DISCUSSION TITLE
what best practice to setup relation on rest api

### DISCUSSION THREAD

**Discussion by rzkmsjhr** [Budibase/budibase#6439] [Help]
I have setup my REST API, to return data that can be shown as a table in Budibase. However, if I have many-to-many relationship table, to show it at Budibase table, what is the best practice to show the correct record in the table since there are more than one reference key that need to be linked?

for example I have 3 table: user, package, booking.

to show, `user name`, `package name`, and `booking date`, I need to put data provider->repeater->data provider->repeater->data provider to show the correct records in the table? what if I have more than 10 tables to reference to?

any help or suggestion will be much appreciated, thanks

**Comment by melohagan** [comment]
Hi @rzkmsjhr 

I'm not sure exactly what the relationships are between the three tables. Would you mind spelling that out please?

It seems in general your approach is correct. With that said, it's probably not ideal to be showing a lot of related data on a single screen.
It may be better to navigate to a new screen passing the ID of the relationship: https://docs.budibase.com/docs/url-parameters
